metro.go

```diff
@@ -93,16 +106,19 @@ func initLogging(to_file bool, level string) {
 	logger, err := log.LoggerFromConfigAsBytes(logConfig)
 	if err != nil {
 		log.Criticalf("Unable to initiate logger: %s", err)
-		os.Exit(1)
+		panic(Exit{1})
 	}
 	log.ReplaceLogger(logger)
+	return logger
 
 }
 
 func main() {
+	defer handleExit()
 	flag.Parse()
 
-	initLogging(true, "warning")
+	logger := initLogging(true, "warning")
+	defer logger.Flush()
```

Comment:
Since you're using `ReplaceLogger` in `initLogging` I think we're good with using `defer log.Flush()` the recommended way. `ReplaceLogger` cleanly replaces the "current" logger (https://github.com/cihub/seelog/blob/5841159c495ad98691aac2bf90113c040e62bac2/log.go#L192) and the `log.Flush()` calls `Flush` on the current logger (https://github.com/cihub/seelog/blob/5841159c495ad98691aac2bf90113c040e62bac2/log.go#L306)